crmsh/report/core.py

```diff
@@ -441,6 +422,38 @@ def load_context_attributes(context: Context) -> None:
     context.cores_dir_list.extend([constants.COROSYNC_LIB] if os.path.isdir(constants.COROSYNC_LIB) else [])
 
 
+def load_context_trace_dir_list(context: Context) -> None:
+    # since the "trace_dir" attribute been removed from cib after untrace
+    # need to parse crmsh log file to extract custom trace ra log directory on each node
+    log_contents = ""
+    cmd = f"grep 'INFO: Trace for .* is written to ' {log.CRMSH_LOG_FILE}*|grep -v 'collect'"
+    for node in context.node_list:
+        shell = context.passwordless_shell_for_nodes.get(node)
+        if shell is not None:
+            ret = shell.subprocess_run_without_input(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        else:
+            # This is a last effort when passwordless ssh is not available
+            cmd = cmd.replace('"', '\\"')
+            ret = _interactive_ssh_run_as_root(node, context.ssh_user, cmd)
+        if ret.returncode == 0:
+            log_contents += ret.stdout.decode('utf-8')
+    context.trace_dir_list = list(set(re.findall("written to (.*)/.*", log_contents)))
+
+
+def _interactive_ssh_run_as_root(host: str, ssh_user: str, cmd: str):
+    # cmd MUST be escaped to make sure it works as an arugment of ssh command
```

Comment:
What does this comment mean for "escaped", in this function scope?

Reply:
The cmd is concatenated to `ssh` as string. This string is firstly parsed by bash and later by ssh. This requires special characters in it to be escaped 2 times. And I am not sure how to write a general function to escape arguments for ssh. So for now, let the callers to escape it case by case.